Content.Client/Verbs/AdminVerbSystem.cs

```diff
@@ -0,0 +1,80 @@
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+
+namespace Content.Client.Verbs
+{
+    /// <summary>
+    ///     Client-side admin verb system. These usually open some sort of UIs.
+    /// </summary>
+    class AdminVerbSystem : EntitySystem
+    {
+        [Dependency] private readonly IClientConGroupController _clientConGroupController = default!;
+        [Dependency] private readonly IViewVariablesManager _viewVariablesManager = default!;
+        [Dependency] private readonly IDynamicTypeFactory _dynamicTypeFactory = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+
+        public override void Initialize()
+        {
+            SubscribeLocalEvent<GetOtherVerbsEvent>(AddAdminVerbs);
+        }
+
+        private void AddAdminVerbs(GetOtherVerbsEvent args)
+        {
+            // View variables verbs
+            if (_clientConGroupController.CanViewVar())
+            {
+                Verb verb = new();
+                verb.Category = VerbCategory.Debug;
+                verb.Text = "View Variables";
+                verb.IconTexture = "/Textures/Interface/VerbIcons/vv.svg.192dpi.png";
+                verb.Act = () => _viewVariablesManager.OpenVV(args.Target);
+                args.Verbs.Add(verb);
+            }
+
+            // Then add some atmos admin verbs that target grid tiles.
+            // However, we do not have verb click coords, and the context menu does not target tiles.
+            // So instead, try find a tile at the clicked entities coordinates:
+            var coords = args.Target.Transform.MapPosition;
+            if (!_mapManager.TryFindGridAt(coords, out var grid))
+                return;
+            var tileIndices = grid.WorldToTile(coords.Position);
+
+            // Verb to open Add Gas admin window, with pre-filled coordinates
+            if (_clientConGroupController.CanCommand("addgas"))
+            {
+                Verb verb = new();
+                verb.Category = VerbCategory.Debug;
+                verb.Text = "Add Gas";
+                // TODO VERB ICON Add gas symbol. A gas tank maybe?
+                verb.Act = () =>
+                {
+                    var window = _dynamicTypeFactory.CreateInstance<AddGasWindow>();
+                    window.FillCoords(grid.Index, tileIndices.X, tileIndices.Y);
+                    window.OpenCentered();
+                };
+                args.Verbs.Add(verb);
+            }
+
+            // Verb to open Set temperature admin window, with pre-filled coordinates
+            if (_clientConGroupController.CanCommand("settemp"))
+            {
+                Verb verb = new();
+                verb.Category = VerbCategory.Debug;
+                verb.Text = "Set Temperature";
+                // TODO VERB ICON add thermometer?
+                verb.Act = () =>
+                {
+                    var window = _dynamicTypeFactory.CreateInstance<SetTemperatureWindow>();
```

Comment:
uhh... did it use dynamictypefactory before? this should be using eUI imo

Reply:
It was doing this before, I just copied it across.  I don't even know why this used dynamic type factory anyways. I don't really know when the type factory would be needed but this function apparently just:
> Constructs a new instance of the given type with Dependencies resolved.

Given that this window has no dependencies, AFAIK this could just be a normal constructor, and it still seems to work fine without the dynamic type factory

Reply:
More details: I got this code from `Content.Client.Administration.UI.CustomControls.UICommandButton`. Previously this window was only opened by pressing this ui button in the admin menu, and I just copied over the same window opening code.

So I guess the reason it was using dynamic type factory is because at least one of those admin menus had dependencies. Now I'm not sure whether to change that code so it no longer uses it or keep it as is.

Reply:
These, and the other new verb can be added sometime later in a seperate PR, removed for now.